src/librustc_metadata/encoder.rs

```diff
@@ -1435,6 +1458,8 @@ impl<'a, 'tcx> EncodeContext<'a, 'tcx> {
             def_path_table: def_path_table,
             impls: impls,
             exported_symbols: exported_symbols,
+            compiler_working_dir: tcx.sess.working_dir.to_string_lossy().into_owned(),
+            debug_prefix_map: debug_prefix_map,
```

Comment:
Yes, that's correct, the mapping and compilation directory will end up in any `rlib` and Rust-`dylib` (but not in executables, `cdylibs`, or `staticlibs`). This is indeed a problem for reproducible builds `:(`